components/board/pi/impl/board.go

```diff
@@ -208,7 +208,7 @@ func (pi *piPigpio) StreamTicks(ctx context.Context, interrupts []board.DigitalI
 	extra map[string]interface{},
 ) error {
 	for _, i := range interrupts {
-		i.AddCallback(ch)
+		AddCallback(i.(*BasicDigitalInterrupt), ch)
```

Comment:
Same question elsewhere, too. `StreamTicks` is a public function, so anyone could call it, and pass in a list of any `DigitalInterrupt`s they want. This could include injected ones in a test, or some custom implementation they've created themselves. I think this line is likely to panic for certain users.

Alas, I don't have a better suggestion of what to do. :confused: If we had a way to limit this to interrupts that the current board created, this line would be in much better shape.

Reply:
My ultimate preference is that we actually have callback implementation be handled by the genericlinux and pi drivers. I don't think the call to make the callbacks function live on the interface rather than driver specific code was a good one, and it was exacerbated by trying to unify the interfaces to account for reconfiguring interrupt pins in the giant reconfigure project. We can fix this another day, though. For now I can live with the panic, it's what was happening anyway when a user trie to call `Tick` `Close` `AddCallback` or `RemoveCallback` from the board interface.

At least right now it has to be:
```go
b, errr := borad.FromDependencies()
pi, ok := b.(pimipl); !ok { err}
di, err := pi.GetDigitalInteruptByName()
dw, ok := di.(*pinwrappers.SomeDigitalInterrupt)
AddCallback(dw) -> panic 
```

We could in the future, demote add-callback and remove callback to only exist in the board implementations. It's more defensive that way, even if repetitive.